CodeSniffer/Standards/Generic/Sniffs/Debug/ESLintSniff.php

```diff
@@ -0,0 +1,147 @@
+ * @link      http://pear.php.net/package/PHP_CodeSniffer
+ */
+class Generic_Sniffs_Debug_ESLintSniff implements PHP_CodeSniffer_Sniff
+{
+    /**
+     * A list of tokenizers this sniff supports.
+     *
+     * @var array
+     */
+    public $supportedTokenizers = array('JS');
+
+
+    /**
+     * ESLint configuration file path.
+     *
+     * @var string|null Path to eslintrc. Null to autodetect.
+     */
+    public $configFile = null;
+
+
+    /**
+     * Returns the token types that this sniff is interested in.
+     *
+     * @return int[]
+     */
+    public function register()
+    {
+        return array(T_OPEN_TAG);
+
+    }//end register()
+
+
+    /**
+     * Processes the tokens that this sniff is interested in.
+     *
+     * @param PHP_CodeSniffer_File $phpcsFile The file where the token was found.
+     * @param int                  $stackPtr  The position in the stack where
+     *                                        the token was found.
+     *
+     * @return void
+     * @throws PHP_CodeSniffer_Exception If jslint.js could not be run
+     */
+    public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
+    {
+        $filename    = $phpcsFile->getFilename();
+        $eslint_path = PHP_CodeSniffer::getConfigData('eslint_path');
+        if ($eslint_path === null) {
+            return;
+        }
+
+        $config_file = $this->configFile;
+        if (empty($config_file) === true) {
+            // Attempt to autodetect.
+            $candidates = glob('.eslintrc{.js,.yaml,.yml,.json}', GLOB_BRACE);
+            if (empty($candidates) === false) {
+                $config_file = $candidates[0];
+            }
+        }
+
+        $eslint_options = array( '--format json' );
+        if (empty($config_file) === false) {
+            $eslint_options[] = sprintf('--config %s', $config_file);
+        }
+
+        $cmd  = sprintf(
+            '"%s" %s "%s"',
+            $eslint_path,
+            implode(' ', $eslint_options),
+            $filename
+        );
+        $desc = array(
+                 0 => array(
+                       'pipe',
+                       'r',
+                      ),
+                 1 => array(
+                       'pipe',
+                       'w',
+                      ),
+                 2 => array(
+                       'pipe',
+                       'w',
+                      ),
+                );
+        $proc = proc_open($cmd, $desc, $pipes);
+
+        // Ignore stdin.
+        fclose($pipes[0]);
+        $stdout = stream_get_contents($pipes[1]);
+        $stderr = stream_get_contents($pipes[2]);
+        fclose($pipes[1]);
+        fclose($pipes[2]);
+
+        // Close, and start working!
+        $code = proc_close($proc);
+
+        if ($code > 0) {
+            $data = json_decode($stdout);
+            // Detect errors.
+            if (json_last_error() !== JSON_ERROR_NONE) {
+                $error = 'Unable to run eslint: %s';
```

Comment:
Sniffs don't add error messages when the external tool fails. Most don't do anything, but one does throw an exception. It's probably best to do nothing so that all the tool sniffs are consistent. In the future, a change could be made to all of them to either throw exceptions or add a warning about a tool failing to run.